go/shared.go

```diff
@@ -1,3 +1,38 @@
 package app
 
+import (
+	"log"
+
+	"go.temporal.io/sdk/workflow"
+)
+
 const BreakfastTaskQueue = "BREAKFAST_TASK_QUEUE"
+
+type Compensations []any
+
+func (s *Compensations) AddCompensation(activity any) {
+	*s = append(*s, activity)
+}
+
+func (s Compensations) Compensate(ctx workflow.Context, in_parallel bool) {
```

Comment:
Go identifiers should not have underscores. This was contested before closing on the Google docs, so I'll quote the canonical docs here, from https://go.dev/doc/effective_go#mixed-caps:

> the convention in Go is to use MixedCaps or mixedCaps rather than underscores to write multiword names.

Reply:
oops, that was my bad. Habits dying hard. I intended to have all MixedCaps/mixedCaps in go.